pydantic/errors.py

```diff
@@ -1,19 +1,46 @@
 from __future__ import annotations as _annotations
 
+from typing import Optional
+
 __all__ = 'PydanticUserError', 'PydanticSchemaGenerationError'
 
 
-class PydanticUserError(TypeError):
+class PydanticErrorMixin:
     """
-    Error caused by incorrect use of pydantic
+    Pydantic Error Mixin for common functions
     """
 
+    def __str__(self) -> str:
+        return f'{self.__class__.__name__}(code={self.code!r}, message={self.message!r})'
+
+    def __repr__(self) -> str:
+        return self.__str__()
+
+    def __init__(self, code: str, *, message: Optional[str] = None) -> None:
```

Comment:
init should come first.

Reply:
Seems like tests keeps failing on tests/test_forward_ref.py

Reply:
Tried adding PydanticErrors to __init__ but it keeps failing as the linter says 
```make
 pydantic\__init__.py:7:21: F401 `.fields.Field` imported but unused
Found 1 error(s).
```

but if I remove Field it says 
```make
E   ImportError: cannot import name 'Field' from 'pydantic' (C:\Users\bgoldstone\Development\pydantic\pydantic\__init__.py)
```

Not sure if you have any suggestions... @samuelcolvin